### DISCUSSION TITLE
Why is this dependency being automatically installed with NPM?

### DISCUSSION THREAD

**Discussion by karljs** [parcel-bundler/parcel#6188] [Q&A]
I'm not a web developer, so bear with me.  I am working on a project that embeds the Ace Editor, and I was planning to just have a copy of the [Ace Builds](https://github.com/ajaxorg/ace-builds) code in my tree.  So I have `ace/ace.js` alongside `index.html`, which includes the line `<script src="./ace/ace.js></script>`.

I do not have a `package.json` as I'm not using NPM to install any dependencies.  I'm using a language that compiles to JavaScript that manages its own dependencies which is not relevant here.

When I run Parcel (version 1.12.5) on `index.html`, it automatically generates a `package.json` which lists `ace-builds` as a dependency, and then it installs that package.  I don't know which version gets bundled in dist.  

I assumed it was parsing this in a weird way and that Parcel didn't realize I wanted a local dependency, but what's weird is if I remove my local copy of `ace/ace.js` I get an `ENOENT` error that it doesn't exist.  I also get an error that the code doesn't exist in `node_modules` if I use `--no-autoinstall`.

So I am being forced to have a local copy of the code and also to let Parcel install the NPM package.  So what's happening here?  Why do I need both?  Really appreciate any help.

**Comment by mischnic** [comment]
Generally, Parcel autoinstalls that package because there is some JS file importing that package.

For example if your `ace.js` file contained
```js
import foo from "ace-builds";
console.log(foo);
```
